sympy/solvers/ode.py

```diff
@@ -1452,6 +1460,37 @@ def _test_term(coeff, order):
     else:
         return tuple(retlist)
 
+def separate_vars(f, symbols=[]):
+    if symbols:
+        if not all((t.is_Atom for t in symbols)):
+            raise ValueError("symbols must be Atoms.")
+    elif symbols is None:
+        return {'coeff':f}
+    else:
+        symbols = list(f.free_symbols)
+        if not symbols:
+            return None
+
+    f = simplify(f)
+    expsym = f.free_symbols
+    intersection = set(symbols).intersection(expsym)
+    if len(intersection) == 0 or len(intersection) == 1:
+        return _separatevars_dict(f, symbols)
+    elif len(intersection) == 2:
+         b = Dummy('b')
+         a = Dummy('a')
+         f1 = f.subs(symbols[1], b)
+         f2 = f.subs(symbols[0], a)
+         f3 = (f.subs(symbols[1], b)).subs(symbols[0], a)
```

Comment:
`f3= f1.subs(symbols[0], a)`

Reply:
Also, would it not be sufficient to see if `not any(i in simplify(expand(cancel(f1*f2/f))).free_symbols for i in symbols)`?

Reply:
yuppp I think it is more general than` simplify(f-f4)` and we can use it.